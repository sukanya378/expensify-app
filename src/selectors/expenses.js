import moment from "moment";

const getVisibleExpenses = (expenses, { text, sortBy, startDate, endDate }) => {
    expenses =  expenses.filter((expense) => {
        const createdAtMoment = moment(expense.createdAt)
        const startDateMatch = startDate ? startDate.isSameOrBefore(createdAtMoment, 'day') : true;
        const endDateMatch = endDate ? endDate.isSameOrAfter(createdAtMoment, 'day') : true;;
        const textMatch = expense.description.toLowerCase().includes(text.toLowerCase())

        return startDateMatch && endDateMatch && textMatch;
    })

    expenses.sort((a,b) => {
        if(sortBy === "amount") return b.amount - a.amount
        else return b.createdAt - a.createdAt
    })
    return expenses
}

export default getVisibleExpenses