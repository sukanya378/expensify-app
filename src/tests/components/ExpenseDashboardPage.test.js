import React from "react";
import { shallow } from "enzyme";
import expenses from "../fixtures/expenses"
import ExpenseDashBoardPage from "../../components/ExpenseDashboardPage";


test('should render Expense dashboard page ', () => {
    const wrapper = shallow(<ExpenseDashBoardPage expenses={expenses} />)
    expect(wrapper).toMatchSnapshot()
})