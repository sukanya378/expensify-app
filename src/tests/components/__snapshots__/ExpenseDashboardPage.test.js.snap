// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Expense dashboard page  1`] = `
<div>
  <Connect(ExpenseListFilter) />
  <Connect(ExpenseList)
    expenses={
      Array [
        Object {
          "amount": 9000,
          "createdAt": 1686126588493,
          "description": "rent",
          "id": "1",
        },
        Object {
          "amount": 200,
          "createdAt": 1686226488493,
          "description": "rent",
          "id": "2",
        },
        Object {
          "amount": 10000,
          "createdAt": 1586226588493,
          "description": "Coffee",
          "id": "3",
        },
        Object {
          "amount": 800,
          "createdAt": 1626226588493,
          "description": "Water Bill",
          "id": "4",
        },
        Object {
          "amount": 1500,
          "createdAt": 1686226588403,
          "description": "Gas Bill",
          "id": "5",
        },
      ]
    }
  />
</div>
`;
